scanner/services/indexer.go

```diff
@@ -0,0 +1,143 @@
+package services
+
+import (
+	"context"
+	"crypto/sha512"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/quay/claircore"
+	"github.com/quay/zlog"
+	v4 "github.com/stackrox/rox/generated/internalapi/scanner/v4"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/pkg/grpc/authz/allow"
+	"github.com/stackrox/rox/scanner/indexer"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type indexerService struct {
+	v4.UnimplementedIndexerServer
+	indexer indexer.Indexer
+}
+
+// NewIndexerService creates a new indexer service.
+func NewIndexerService(indexer indexer.Indexer) *indexerService {
+	return &indexerService{
+		indexer: indexer,
+	}
+}
+
+func (s *indexerService) CreateIndexReport(ctx context.Context, req *v4.CreateIndexReportRequest) (*v4.IndexReport, error) {
+	// TODO We currently only support container images, hence we assume the resource
+	//      is of that type. When introducing nodes and other resources, this should
+	//      evolve.
+	resourceType := "containerimage"
+	if err := validateContainerImageRequest(req); err != nil {
+		return nil, err
+	}
+	// Create a claircore.Digest using the full Hash ID, including resource type.
+	hashIDSum := sha512.Sum512([]byte(req.GetHashId()))
```

Comment:
When sensor calls CreateIndexReport() api, does it need to provide HashId? I thought hashId is generated with the index report in Claircore?